Files/Extensions/DateTimeExtensions.cs

```diff
@@ -66,15 +66,16 @@ public static (string text, string range, string glyph, int index) GetFriendlyTi
             Windows.Globalization.Calendar cal = new Windows.Globalization.Calendar();
             var t = DateTimeOffset.Now;
             var t2 = dt.ToLocalTime();
+            var y = t.AddDays(-1);
             var today = DateTime.Today;
 
             var diff = t - dt;
-            if (t.Month == t2.Month && t.Day == t2.Day)
+            if (t.Month == t2.Month && t.Day == t2.Day && t.Year == t2.Year)
```

Comment:
looks good. can you please add this fix to the other two checks below this one (line 83 and 88)?

Reply:
thanks, I will, 
I will also fix line 98 which I think will break with year change (Jan - 1 => -11)

Reply:
@winston-de  I just realised, on line 83, "This Week" can be partially in two years (ie. this year Mon-Fri is in 2021 and Sat-Sun is 2022) so maybe the year check shouldn't be there. diff.Days should ensure it won't be showing files older than 7 days anyway, right?). Need to run some checks what GetWeekOfYear returns depending if it's 31st or 1st (but same week) etc

Similarly line 88: "Last Week" could be in a previous year, however this one needs another fix regardless for year change, ie. when 't2' week is in new year (week 1) and previous one is week 52.

Reply:
Good catch, a solution could be to calculate the number of weeks since the default date, and use that to check if the weeks are the same.

Reply:
OK, just pushed some more changes to the date grouping, using variable "w" for week comparisons X days ago (where X is the days difference). This should correctly handle different culture/calendar setups

And small fix for LastMonth check as well.

I hope that fixes it all :)